### ISSUE TITLE
Chinese text missing

### ISSUE THREAD

**Issue by JO-Step** [ruffle-rs/ruffle#7164]
### Describe the bug

All Chinese characters are not shown on the link or local file
http://www.maaak.net/mting.swf in http://www.maaak.net/mt2_play.php

### Expected behavior

Display the characters 

### Affected platform

Desktop app

### Operating system

Win10

**Comment by n0samu** [comment]
It took me a bit to find the problem, so to save the developers some time, here's a demonstration.
Flash Player:
![image](https://user-images.githubusercontent.com/71368227/172041071-a16b747e-cab7-4405-8aba-a7319ef5193d.png)
Ruffle:
![image](https://user-images.githubusercontent.com/71368227/172041183-f8a91ee2-1e46-4900-a8e6-4ce83c305537.png)

Anyway, this usually happens when an SWF does not embed all of the fonts that it needs. Flash Player would fall back to a device font in this case, but Ruffle's fake device font does not support Chinese characters. See #1862.

**Comment by CatMe0w** [comment]
@n0samu Hi. I want to let you know this issue seems to be fixed in Windows builds during https://github.com/ruffle-rs/ruffle/compare/nightly-2024-01-08...nightly-2024-01-09

![screenshot](https://github.com/ruffle-rs/ruffle/assets/7774491/40ef22b1-d0e2-4399-8fd9-cf9a6126d9f8)

Left: Ruffle nightly 2014-01-08
Middle: Ruffle nightly 2014-01-09
Right: Flash 32.0.0.465

But as you can see, the italicized text is not rendered properly, a problem that still exists in the current builds (nightly 2024-04-10).

Also, other platforms still have this issue.

![screenshot2](https://github.com/ruffle-rs/ruffle/assets/7774491/51bb1819-025d-4ff7-b0da-7201812588f8)

Left: Ruffle nightly 2024-04-10, Linux; Completely cannot see any text including ASCII characters.
Right: Ruffle nightly 2024-04-10, web extension; Can only see ASCII characters and italicized text is not working.
Note that web extensions have the same behaviors on all OS and browsers.

![screenshot3](https://github.com/ruffle-rs/ruffle/assets/7774491/406697ba-2a10-4fd7-ae43-af2d89068e3b)

Ruffle nightly 2024-04-10, macOS; Can only see ASCII characters and italicized text is working (This is not shown in this screenshot).

The swf I tested can be found at `https://catme0w[.]org/xmdfw/play/`. The swf embedded the fonts its used.
(This is a Monooply-like game. To start the game, click on the button on the first line "开始", pick 4 characters and click on the bottom right button "开始游戏". After starting the game, click on the blinking "StarQ" to roll the dice.)

**Comment by n0samu** [comment]
@CatMe0w Please open a new issue. The problems with your game are probably not related to the game originally mentioned in this issue.

**Comment by CatMe0w** [comment]
Sorry! I thought they were the same problem but I just confirmed they're not. 

Status: `mting.swf` is still not rendering text properly after #14535.

**Comment by kjarosh** [comment]
No way of reproducing this issue, the website is down

**Comment by CatMe0w** [comment]
Internet Archive has it: https://web.archive.org/web/20210310004316/http://www.maaak.net/mting.swf
